packages/chat/src/conversation/system.ts

```diff
@@ -0,0 +1,37 @@
+import { Conversation, uuid } from '@botpress/messaging-base'
+import { MessagingSocket } from '@botpress/messaging-socket'
+import { WebchatStorage } from '../storage/system'
+import { ConversationEmitter, ConversationEvents, ConversationWatcher } from './events'
+
+export class WebchatConversation {
+  public readonly events: ConversationWatcher
+  private readonly emitter: ConversationEmitter
+  private current?: Conversation
+
+  constructor(private storage: WebchatStorage, private socket: MessagingSocket) {
+    this.emitter = new ConversationEmitter()
+    this.events = this.emitter
+  }
+
+  async setup() {
+    const saved = this.storage.get<uuid>('saved-conversation')
+
+    const event = { choice: saved }
+    await this.emitter.emit(ConversationEvents.Choose, event)
+
+    const conversation = await this.socket.conversations.use(event.choice)
```

Comment:
Yes use will always return a conversation even if you provided a conversation id that does not exist or undefined. This might seem conterintuitive but it's done this way to reduce the amount of back and forth between the client and the server, since if we got not conversation back we would ask the server to create one anyway. 

Most of the calls that we do here in the webhooks are likely going to be changed to a single request that sets up everything to even further the amount of back and forth. We can still implement clear api endpoints but think of these endpoints as optimized for the most common use case.